clang/include/clang/AST/ASTNodeTraverser.h

```diff
@@ -149,9 +149,13 @@ class ASTNodeTraverser
 
       for (const Stmt *SubStmt : S->children())
         Visit(SubStmt);
+
```

Comment:
`ConstStmtVisitor<Derived>::Visit(S)` (line 141) is dispatched to `ASTDumper::VisitSomeStmt` if it's defined.
However, visiting the node occurs before visiting its `children()`.

If we add where-clauses dump in the `ASTDumper::VisitSomeStmt`, in the dump result, the node for where is before the node for its children nodes. e.g. dumping `i = 1 where x = 1` will have
```
Binop `=`
|--Where
|----EqFact
|--------Fact (x=1)
|--Var i
|--IntLiteral 1
```
it's not consistent with the sequence of tokens in the source code. The better dumping result is

```
Binop `=`
|--Var i
|--IntLiteral 1
|--Where
|----EqFact
|--------Fact (x=1)
```